sktime/transformations/series/detrend/_deseasonalize.py

```diff
@@ -316,3 +322,223 @@ def fit(self, Z, X=None):
+        Length of the seasonal smoother. Must be an odd integer, and should
+        normally be >= 7 (default).
+    trend : {int, default=None}
+        Length of the trend smoother. Must be an odd integer. If not provided
+        uses the smallest odd integer greater than
+        1.5 * period / (1 - 1.5 / seasonal), following the suggestion in
+        the original implementation.
+    low_pass : {int, default=None}
+        Length of the low-pass filter. Must be an odd integer >=3. If not
+        provided, uses the smallest odd integer > period.
+    seasonal_deg : int, default=1
+        Degree of seasonal LOESS. 0 (constant) or 1 (constant and trend).
+    trend_deg : int, default=1
+        Degree of trend LOESS. 0 (constant) or 1 (constant and trend).
+    low_pass_deg : int, default=1
+        Degree of low pass LOESS. 0 (constant) or 1 (constant and trend).
+    robust : bool, default False
+        Flag indicating whether to use a weighted version that is robust to
+        some forms of outliers.
+    seasonal_jump : int, default=1
+        Positive integer determining the linear interpolation step. If larger
+        than 1, the LOESS is used every seasonal_jump points and linear
+        interpolation is between fitted points. Higher values reduce
+        estimation time.
+    trend_jump : int, default=1
+        Positive integer determining the linear interpolation step. If larger
+        than 1, the LOESS is used every trend_jump points and values between
+        the two are linearly interpolated. Higher values reduce estimation
+        time.
+    low_pass_jump : int, default=1
+        Positive integer determining the linear interpolation step. If larger
+        than 1, the LOESS is used every low_pass_jump points and values between
+        the two are linearly interpolated. Higher values reduce estimation
+        time.
+
+    Attributes
+    ----------
+    seasonal : array of length sp
+        Seasonal components computed in seasonal decomposition.
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.compose import TransformedTargetForecaster
+    >>> from sktime.forecasting.trend import PolynomialTrendForecaster
+    >>> import numpy as np
+    >>> y = load_airline()
+    >>> transformer = STL(sp=12)
+    >>> y_hat = transformer.fit_transform(y)
+    >>> forecaster = TransformedTargetForecaster([
+    >>>     ("deseasonalize", STL(sp=12, seasonal=13, seasonal_deg=0)),
+    >>>     ("forecast", PolynomialTrendForecaster())]
+    >>> )
+    >>> forecaster.fit(y)
+    >>> y_pred = forecaster.predict(fh=np.arange(1,13))
+    """
+
+    _tags = {"transform-returns-same-time-index": True, "univariate-only": True}
+
+    def __init__(self, sp=1, seasonal=7, trend=None, low_pass=None,
+                 seasonal_deg=1, trend_deg=1, low_pass_deg=1,
+                 robust=False, seasonal_jump=1, trend_jump=1, low_pass_jump=1):
+        self.sp = check_sp(sp)
```

Comment:
I htink its best to do that check in `fit()`

Reply:
I just followed the Deseasonalizer code for this one. I can go either way but I think it makes sense to check the seasonality parameter when initializing the algorithm and not wait for the fit. I'm happy to move it to fit.